web/src/lightbox.ts

```diff
@@ -205,11 +205,13 @@ export function clear_for_testing(): void {
 }
 
 export function canonical_url_of_media(media: HTMLElement): string {
-    let media_src = media.getAttribute("src");
-    if (!media_src || media_src.startsWith("/user_uploads/thumbnail/")) {
-        media_src = media.parentElement!.getAttribute("href")!;
+    let media_url = "";
```

Comment:
I was angering Typescript, so I'm not at all confident about how I've done this. But checking for `src` and then `href` on their respective HTML element types seemed a reasonable way to proceed.

Reply:
We want to use the parent href if the image is a thumbnail, still.  That logic looks to have been lost

Reply:
Got it. And restored. Is that a legacy thing? Like, newly uploaded `.mp4` files don't see to have a thumbnail image.